internal/manifest/version.go

```diff
@@ -289,15 +297,26 @@ func (v *Version) Overlaps(
 					end = largest
 					restart = true
 				}
-				if restart {
-					ret = ret[:0]
-					continue loop
+			}
+
+			if restart {
+				// Need to retry the files we have not selected.
+				restart = false
+			} else {
+				// TODO: is there a better way to reserve numSelected in ret?
+				ret = append(ret, make([]FileMetadata, numSelected)...)
```

Comment:
Yeah, the more idiomatic way is to create a slice with capacity but 0-length:

```
ret = make([]FileMetadata, 0, numSelected)
```

Then in the loop you append to the slice which allows you to omit `j`:

```
for i, selected := range selectedIndices {
  if selected {
    ret = append(ret, v.Files[level][i])
  }
}
```

Reply:
done